src/libmeasurement_kit/dns/query.cpp

```diff
@@ -15,7 +16,14 @@ void query(
         Settings settings,
         Var<Reactor> reactor,
         Var<Logger> logger) {
-    libevent::query(dns_class, dns_type, name, cb, settings, reactor, logger);
+    std::string engine = settings.get("dns/engine", std::string("libevent"));
+    if (engine == "libevent") {
+        libevent::query(dns_class, dns_type, name, cb, settings, reactor, logger);
+    } else if (engine == "system") {
+        system_resolver(dns_class, dns_type, name, cb, settings, reactor, logger);
+    } else {
+        cb(InvalidDnsEngine(), nullptr);
```

Comment:
(rationale: I hate when libraries call *callbacks* ahead in time. I suppose callbacks should be called later.